File: src/main/java/cn/cloudself/model/DeliverEntity.java
package cn.cloudself.model;

import org.hibernate.annotations.DynamicInsert;

import javax.persistence.*;

/**
 * @author HerbLuo
 * @version 1.0.0.d
 */
@Entity
@DynamicInsert
@Table(name = "deliver", schema = "shop")
public class DeliverEntity {
    private int id;

    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    private String number;

    @Basic
    @Column(name = "number", nullable = true, length = 64)
    public String getNumber() {
        return number;
    }    private String name;

    public void setNumber(String number) {
        this.number = number;
    }

    @Basic
    @Column(name = "name", nullable = true, length = 10)
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    private Integer orderAShopId;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        DeliverEntity that = (DeliverEntity) o;

        if (id != that.id) return false;
        if (number != null ? !number.equals(that.number) : that.number != null) return false;
        if (name != null ? !name.equals(that.name) : that.name != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + (number != null ? number.hashCode() : 0);
        result = 31 * result + (name != null ? name.hashCode() : 0);
        return result;
    }

    @Basic
    @Column(name = "order_a_shop_id")
    public Integer getOrderAShopId() {
        return orderAShopId;
    }

    public void setOrderAShopId(Integer orderAShopId) {
        this.orderAShopId = orderAShopId;
    }
}

File: src/main/java/cn/cloudself/model/DeliverEntity.kt
package cn.cloudself.model

import org.hibernate.annotations.DynamicInsert

import javax.persistence.*

/**
 * @author HerbLuo
 * @version 1.0.0.d
 */
@Entity
@DynamicInsert
@Table(name = "deliver", schema = "shop")
data class DeliverEntity(
        @get:Id
        @get:Column(name = "id", nullable = false)
        @get:GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Int = 0,

        @get:Basic
        @get:Column(name = "number", nullable = true, length = 64)
        var number: String? = null,

        @get:Basic
        @get:Column(name = "name", nullable = true, length = 10)
        var name: String? = null,

        @get:Basic
        @get:Column(name = "order_a_shop_id")
        var orderAShopId: Int? = null
)
